{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "d13f8e99-9799-4367-ba6f-77553852425c",
   "metadata": {},
   "outputs": [],
   "source": [
    "import esm\n",
    "import torch\n",
    "import time\n",
    "import gc\n",
    "import os\n",
    "import numpy as np\n",
    "from Bio import SeqIO\n",
    "import pandas as pd\n",
    "from cyvcf2 import VCF\n",
    "\n",
    "%run \"../scripts/data_processing.py\"\n",
    "%run \"../scripts/node_edge_generation.py\"\n",
    "%run \"../scripts/graph_functions.py\"\n",
    "\n",
    "#Data path\n",
    "datadir = os.path.abspath(\"../../Data/\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "7eee9a3b-9c24-4e0c-bfaf-9f11b5524219",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 38792996\n",
      "-rwxr-xr-x 1 louie louie  134883391 Sep 16 14:44 SRR28808391.fastq.gz\n",
      "-rwxr-xr-x 1 louie louie  175829790 Sep 16 14:47 SRR28808366.fastq.gz\n",
      "-rwxr-xr-x 1 louie louie  137556987 Sep 16 14:47 SRR28808369.fastq.gz\n",
      "-rwxr-xr-x 1 louie louie  125930394 Sep 16 14:47 SRR28808379.fastq.gz\n",
      "-rwxr-xr-x 1 louie louie  208640818 Sep 16 14:48 SRR28808393.fastq.gz\n",
      "-rwxr-xr-x 1 louie louie  216681502 Sep 16 14:48 SRR28808373.fastq.gz\n",
      "-rwxr-xr-x 1 louie louie  251263758 Sep 16 14:48 SRR28808365.fastq.gz\n",
      "-rwxr-xr-x 1 louie louie  298368113 Sep 16 14:48 SRR28808367.fastq.gz\n",
      "-rwxr-xr-x 1 louie louie  256624158 Sep 16 14:48 SRR28808372.fastq.gz\n",
      "-rwxr-xr-x 1 louie louie  270419069 Sep 16 14:48 SRR28808368.fastq.gz\n",
      "-rwxr-xr-x 1 louie louie  215636846 Sep 16 14:48 SRR28808370.fastq.gz\n",
      "-rwxr-xr-x 1 louie louie  199309783 Sep 16 14:48 SRR28808376.fastq.gz\n",
      "-rwxr-xr-x 1 louie louie  157074821 Sep 16 14:51 SRR28808380.fastq.gz\n",
      "-rwxr-xr-x 1 louie louie  139678662 Sep 16 14:51 SRR28808392.fastq.gz\n",
      "-rwxr-xr-x 1 louie louie  310224278 Sep 16 14:51 SRR28808375.fastq.gz\n",
      "-rwxr-xr-x 1 louie louie  176720231 Sep 16 14:51 SRR28808384.fastq.gz\n",
      "-rwxr-xr-x 1 louie louie  173114958 Sep 16 14:51 SRR28808388.fastq.gz\n",
      "-rwxr-xr-x 1 louie louie  210053006 Sep 16 14:51 SRR28808383.fastq.gz\n",
      "-rwxr-xr-x 1 louie louie  212236038 Sep 16 14:51 SRR28808374.fastq.gz\n",
      "-rwxr-xr-x 1 louie louie  215952390 Sep 16 14:51 SRR28808364.fastq.gz\n",
      "-rwxr-xr-x 1 louie louie  142855379 Sep 16 14:52 SRR28808377.fastq.gz\n",
      "-rwxr-xr-x 1 louie louie  288486594 Sep 16 14:52 SRR28808371.fastq.gz\n",
      "-rwxr-xr-x 1 louie louie  144572713 Sep 16 14:53 SRR28808378.fastq.gz\n",
      "-rwxr-xr-x 1 louie louie  148316994 Sep 16 14:53 SRR28808381.fastq.gz\n",
      "-rwxr-xr-x 1 louie louie  154305856 Sep 16 14:53 SRR28808386.fastq.gz\n",
      "-rwxr-xr-x 1 louie louie  153834877 Sep 16 14:53 SRR28808390.fastq.gz\n",
      "-rwxr-xr-x 1 louie louie  174920929 Sep 16 14:53 SRR28808387.fastq.gz\n",
      "-rwxr-xr-x 1 louie louie  166481856 Sep 16 14:53 SRR28808385.fastq.gz\n",
      "-rwxr-xr-x 1 louie louie  183160007 Sep 16 14:53 SRR28808389.fastq.gz\n",
      "-rwxr-xr-x 1 louie louie  166674789 Sep 16 14:53 SRR28808382.fastq.gz\n",
      "-rwxr-xr-x 1 louie louie 1248241184 Sep 16 15:09 SRR28808364.fastq\n",
      "-rwxr-xr-x 1 louie louie 1457405007 Sep 16 15:09 SRR28808365.fastq\n",
      "-rwxr-xr-x 1 louie louie 1018615597 Sep 16 15:09 SRR28808366.fastq\n",
      "-rwxr-xr-x 1 louie louie 1807645190 Sep 16 15:10 SRR28808367.fastq\n",
      "-rwxr-xr-x 1 louie louie 1597499184 Sep 16 15:10 SRR28808368.fastq\n",
      "-rwxr-xr-x 1 louie louie  784056505 Sep 16 15:10 SRR28808369.fastq\n",
      "-rwxr-xr-x 1 louie louie 1234309506 Sep 16 15:10 SRR28808370.fastq\n",
      "-rwxr-xr-x 1 louie louie 1697102532 Sep 16 15:10 SRR28808371.fastq\n",
      "-rwxr-xr-x 1 louie louie 1490016382 Sep 16 15:10 SRR28808372.fastq\n",
      "-rwxr-xr-x 1 louie louie 1257204486 Sep 16 15:10 SRR28808373.fastq\n",
      "-rwxr-xr-x 1 louie louie 1246177557 Sep 16 15:10 SRR28808374.fastq\n",
      "-rwxr-xr-x 1 louie louie 1900941138 Sep 16 15:10 SRR28808375.fastq\n",
      "-rwxr-xr-x 1 louie louie 1162716707 Sep 16 15:10 SRR28808376.fastq\n",
      "-rwxr-xr-x 1 louie louie  846904367 Sep 16 15:10 SRR28808377.fastq\n",
      "-rwxr-xr-x 1 louie louie  862543090 Sep 16 15:11 SRR28808378.fastq\n",
      "-rwxr-xr-x 1 louie louie  727658785 Sep 16 15:11 SRR28808379.fastq\n",
      "-rwxr-xr-x 1 louie louie  907690887 Sep 16 15:11 SRR28808380.fastq\n",
      "-rwxr-xr-x 1 louie louie  837520181 Sep 16 15:11 SRR28808381.fastq\n",
      "-rwxr-xr-x 1 louie louie  963208915 Sep 16 15:11 SRR28808382.fastq\n",
      "-rwxr-xr-x 1 louie louie 1205484808 Sep 16 15:11 SRR28808383.fastq\n",
      "-rwxr-xr-x 1 louie louie 1026407775 Sep 16 15:11 SRR28808384.fastq\n",
      "-rwxr-xr-x 1 louie louie  969766632 Sep 16 15:11 SRR28808385.fastq\n",
      "-rwxr-xr-x 1 louie louie  916233371 Sep 16 15:11 SRR28808386.fastq\n",
      "-rwxr-xr-x 1 louie louie 1007735556 Sep 16 15:11 SRR28808387.fastq\n",
      "-rwxr-xr-x 1 louie louie 1001660067 Sep 16 15:11 SRR28808388.fastq\n",
      "-rwxr-xr-x 1 louie louie 1060928633 Sep 16 15:11 SRR28808389.fastq\n",
      "-rwxr-xr-x 1 louie louie  874813008 Sep 16 15:11 SRR28808390.fastq\n",
      "-rwxr-xr-x 1 louie louie  792439186 Sep 16 15:11 SRR28808391.fastq\n",
      "-rwxr-xr-x 1 louie louie  808512001 Sep 16 15:11 SRR28808392.fastq\n",
      "-rwxr-xr-x 1 louie louie 1202374081 Sep 16 15:12 SRR28808393.fastq\n",
      "drwxrwxr-x 2 louie louie       4096 Sep 18 16:50 SRR28808369\n",
      "drwxrwxr-x 2 louie louie       4096 Sep 18 16:50 SRR28808364\n",
      "drwxrwxr-x 2 louie louie       4096 Sep 18 16:50 SRR28808393\n",
      "drwxrwxr-x 2 louie louie       4096 Sep 18 16:50 SRR28808388\n",
      "drwxrwxr-x 2 louie louie       4096 Sep 18 16:50 SRR28808378\n",
      "drwxrwxr-x 2 louie louie       4096 Sep 18 16:50 SRR28808381\n",
      "drwxrwxr-x 2 louie louie       4096 Sep 18 16:50 SRR28808372\n",
      "drwxrwxr-x 2 louie louie       4096 Sep 18 16:50 SRR28808392\n",
      "drwxrwxr-x 2 louie louie       4096 Sep 18 16:50 SRR28808389\n",
      "drwxrwxr-x 2 louie louie       4096 Sep 18 16:50 SRR28808383\n",
      "drwxrwxr-x 2 louie louie       4096 Sep 18 16:50 SRR28808366\n",
      "drwxrwxr-x 2 louie louie       4096 Sep 18 17:47 SRR28808386\n",
      "drwxrwxr-x 2 louie louie       4096 Sep 18 17:59 SRR28808390\n",
      "drwxrwxr-x 2 louie louie       4096 Sep 18 17:59 SRR28808376\n",
      "drwxrwxr-x 2 louie louie       4096 Sep 18 17:59 SRR28808368\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "os.system(\"ls -ltr \"+ datadir+\"/GWAS/PRJNA1104541\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "baa6b7ef-6e80-4d66-b89d-aeb5a2a4581c",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[bwa_index] Pack FASTA... 9.49 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[BWTIncCreate] textLength=6199501436, availableWord=448219384\n",
      "[BWTIncConstructFromPacked] 10 iterations done. 99999996 characters processed.\n",
      "[BWTIncConstructFromPacked] 20 iterations done. 199999996 characters processed.\n",
      "[BWTIncConstructFromPacked] 30 iterations done. 299999996 characters processed.\n",
      "[BWTIncConstructFromPacked] 40 iterations done. 399999996 characters processed.\n",
      "[BWTIncConstructFromPacked] 50 iterations done. 499999996 characters processed.\n",
      "[BWTIncConstructFromPacked] 60 iterations done. 599999996 characters processed.\n",
      "[BWTIncConstructFromPacked] 70 iterations done. 699999996 characters processed.\n",
      "[BWTIncConstructFromPacked] 80 iterations done. 799999996 characters processed.\n",
      "[BWTIncConstructFromPacked] 90 iterations done. 899999996 characters processed.\n",
      "[BWTIncConstructFromPacked] 100 iterations done. 999999996 characters processed.\n",
      "[BWTIncConstructFromPacked] 110 iterations done. 1099999996 characters processed.\n",
      "[BWTIncConstructFromPacked] 120 iterations done. 1199999996 characters processed.\n",
      "[BWTIncConstructFromPacked] 130 iterations done. 1299999996 characters processed.\n",
      "[BWTIncConstructFromPacked] 140 iterations done. 1399999996 characters processed.\n",
      "[BWTIncConstructFromPacked] 150 iterations done. 1499999996 characters processed.\n",
      "[BWTIncConstructFromPacked] 160 iterations done. 1599999996 characters processed.\n",
      "[BWTIncConstructFromPacked] 170 iterations done. 1699999996 characters processed.\n",
      "[BWTIncConstructFromPacked] 180 iterations done. 1799999996 characters processed.\n",
      "[BWTIncConstructFromPacked] 190 iterations done. 1899999996 characters processed.\n",
      "[BWTIncConstructFromPacked] 200 iterations done. 1999999996 characters processed.\n",
      "[BWTIncConstructFromPacked] 210 iterations done. 2099999996 characters processed.\n",
      "[BWTIncConstructFromPacked] 220 iterations done. 2199999996 characters processed.\n",
      "[BWTIncConstructFromPacked] 230 iterations done. 2299999996 characters processed.\n",
      "[BWTIncConstructFromPacked] 240 iterations done. 2399999996 characters processed.\n",
      "[BWTIncConstructFromPacked] 250 iterations done. 2499999996 characters processed.\n",
      "[BWTIncConstructFromPacked] 260 iterations done. 2599999996 characters processed.\n",
      "[BWTIncConstructFromPacked] 270 iterations done. 2699999996 characters processed.\n",
      "[BWTIncConstructFromPacked] 280 iterations done. 2799999996 characters processed.\n",
      "[BWTIncConstructFromPacked] 290 iterations done. 2899999996 characters processed.\n",
      "[BWTIncConstructFromPacked] 300 iterations done. 2999999996 characters processed.\n",
      "[BWTIncConstructFromPacked] 310 iterations done. 3099999996 characters processed.\n",
      "[BWTIncConstructFromPacked] 320 iterations done. 3199999996 characters processed.\n",
      "[BWTIncConstructFromPacked] 330 iterations done. 3299999996 characters processed.\n",
      "[BWTIncConstructFromPacked] 340 iterations done. 3399999996 characters processed.\n",
      "[BWTIncConstructFromPacked] 350 iterations done. 3499999996 characters processed.\n",
      "[BWTIncConstructFromPacked] 360 iterations done. 3599999996 characters processed.\n",
      "[BWTIncConstructFromPacked] 370 iterations done. 3699999996 characters processed.\n",
      "[BWTIncConstructFromPacked] 380 iterations done. 3799999996 characters processed.\n",
      "[BWTIncConstructFromPacked] 390 iterations done. 3899999996 characters processed.\n",
      "[BWTIncConstructFromPacked] 400 iterations done. 3999999996 characters processed.\n",
      "[BWTIncConstructFromPacked] 410 iterations done. 4099999996 characters processed.\n",
      "[BWTIncConstructFromPacked] 420 iterations done. 4199999996 characters processed.\n",
      "[BWTIncConstructFromPacked] 430 iterations done. 4299999996 characters processed.\n",
      "[BWTIncConstructFromPacked] 440 iterations done. 4399999996 characters processed.\n",
      "[BWTIncConstructFromPacked] 450 iterations done. 4499999996 characters processed.\n",
      "[BWTIncConstructFromPacked] 460 iterations done. 4599999996 characters processed.\n",
      "[BWTIncConstructFromPacked] 470 iterations done. 4699999996 characters processed.\n",
      "[BWTIncConstructFromPacked] 480 iterations done. 4799999996 characters processed.\n",
      "[BWTIncConstructFromPacked] 490 iterations done. 4899999996 characters processed.\n",
      "[BWTIncConstructFromPacked] 500 iterations done. 4999999996 characters processed.\n",
      "[BWTIncConstructFromPacked] 510 iterations done. 5099999996 characters processed.\n",
      "[BWTIncConstructFromPacked] 520 iterations done. 5199999996 characters processed.\n",
      "[BWTIncConstructFromPacked] 530 iterations done. 5299999996 characters processed.\n",
      "[BWTIncConstructFromPacked] 540 iterations done. 5399999996 characters processed.\n",
      "[BWTIncConstructFromPacked] 550 iterations done. 5499999996 characters processed.\n",
      "[BWTIncConstructFromPacked] 560 iterations done. 5596733548 characters processed.\n",
      "[BWTIncConstructFromPacked] 570 iterations done. 5683010028 characters processed.\n",
      "[BWTIncConstructFromPacked] 580 iterations done. 5759688796 characters processed.\n",
      "[BWTIncConstructFromPacked] 590 iterations done. 5827837068 characters processed.\n",
      "[BWTIncConstructFromPacked] 600 iterations done. 5888403388 characters processed.\n",
      "[BWTIncConstructFromPacked] 610 iterations done. 5942230876 characters processed.\n",
      "[BWTIncConstructFromPacked] 620 iterations done. 5990068876 characters processed.\n",
      "[BWTIncConstructFromPacked] 630 iterations done. 6032583388 characters processed.\n",
      "[BWTIncConstructFromPacked] 640 iterations done. 6070366396 characters processed.\n",
      "[BWTIncConstructFromPacked] 650 iterations done. 6103943980 characters processed.\n",
      "[BWTIncConstructFromPacked] 660 iterations done. 6133783772 characters processed.\n",
      "[BWTIncConstructFromPacked] 670 iterations done. 6160301452 characters processed.\n",
      "[BWTIncConstructFromPacked] 680 iterations done. 6183866396 characters processed.\n",
      "[bwt_gen] Finished constructing BWT in 688 iterations.\n",
      "[bwa_index] 1442.49 seconds elapse.\n",
      "[bwa_index] Update BWT... 8.42 sec\n",
      "[bwa_index] Pack forward-only FASTA... 5.56 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... "
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Reference index complete | 36.07257577578227 minutes\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "692.61 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index /home/louie/Projects/Personal/GraphDB/Data/GWAS/Hsapiens_ref/ENSEMBL_GRCh38.p14/Homo_sapiens.GRCh38.dna_sm.primary_assembly.fa\n",
      "[main] Real time: 2164.351 sec; CPU: 2158.586 sec\n"
     ]
    }
   ],
   "source": [
    "#reference = datadir+\"/GWAS/Hsapiens_ref/NCBI_GRCh38.p14/GCF_000001405.40_GRCh38.p14_genomic.fna\"\n",
    "reference = datadir+\"/GWAS/Hsapiens_ref/ENSEMBL_GRCh38.p14/Homo_sapiens.GRCh38.dna_sm.primary_assembly.fa\"\n",
    "start = time.time()\n",
    "# Index the reference genome\n",
    "os.system(f'bwa index {reference}')\n",
    "print(f'Reference index complete | {(time.time()-start)/60} minutes')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "ecaba772-4890-43a5-a676-b41b10fd08ed",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "~~~~~~~~~~\n",
      "(1) - Main file - SRR28808386.fastq\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 1682952 sequences (120000050 bp)...\n",
      "[M::process] read 1681744 sequences (120000007 bp)...\n",
      "[M::mem_process_seqs] Processed 1682952 reads in 439.702 CPU sec, 36.785 real sec\n",
      "[M::process] read 1020803 sequences (72939405 bp)...\n",
      "[M::mem_process_seqs] Processed 1681744 reads in 452.644 CPU sec, 37.817 real sec\n",
      "[M::mem_process_seqs] Processed 1020803 reads in 274.654 CPU sec, 22.919 real sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa mem -t 12 /home/louie/Projects/Personal/GraphDB/Data/GWAS/Hsapiens_ref/ENSEMBL_GRCh38.p14/Homo_sapiens.GRCh38.dna_sm.primary_assembly.fa /home/louie/Projects/Personal/GraphDB/Data/GWAS/PRJNA1104541/SRR28808386/SRR28808386.fastq\n",
      "[main] Real time: 101.531 sec; CPU: 1169.784 sec\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[mpileup] 1 samples in 1 input files\n",
      "[mpileup] maximum number of reads per input file set to -d 250\n",
      "Note: none of --samples-file, --ploidy or --ploidy-file given, assuming all sites are diploid\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SRR28808386.fastq analysis complete | 3.7725327372550965 minutes\n",
      "\n",
      "\n",
      "~~~~~~~~~~\n",
      "(2) - Main file - SRR28808390.fastq\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 1656520 sequences (120000008 bp)...\n",
      "[M::process] read 1655932 sequences (120000057 bp)...\n",
      "[M::mem_process_seqs] Processed 1656520 reads in 433.221 CPU sec, 36.272 real sec\n",
      "[M::process] read 830676 sequences (60279008 bp)...\n",
      "[M::mem_process_seqs] Processed 1655932 reads in 427.793 CPU sec, 35.731 real sec\n",
      "[M::mem_process_seqs] Processed 830676 reads in 218.493 CPU sec, 18.213 real sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa mem -t 12 /home/louie/Projects/Personal/GraphDB/Data/GWAS/Hsapiens_ref/ENSEMBL_GRCh38.p14/Homo_sapiens.GRCh38.dna_sm.primary_assembly.fa /home/louie/Projects/Personal/GraphDB/Data/GWAS/PRJNA1104541/SRR28808390/SRR28808390.fastq\n",
      "[main] Real time: 93.063 sec; CPU: 1081.961 sec\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[mpileup] 1 samples in 1 input files\n",
      "[mpileup] maximum number of reads per input file set to -d 250\n",
      "Note: none of --samples-file, --ploidy or --ploidy-file given, assuming all sites are diploid\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SRR28808390.fastq analysis complete | 3.1317101319630942 minutes\n",
      "\n",
      "\n",
      "~~~~~~~~~~\n",
      "(3) - Main file - SRR28808376.fastq\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 1668328 sequences (120000107 bp)...\n",
      "[M::process] read 1666396 sequences (120000128 bp)...\n",
      "[M::mem_process_seqs] Processed 1668328 reads in 392.097 CPU sec, 32.789 real sec\n",
      "[M::process] read 1668092 sequences (120000020 bp)...\n",
      "[M::mem_process_seqs] Processed 1666396 reads in 401.251 CPU sec, 33.481 real sec\n",
      "[M::process] read 528178 sequences (38112578 bp)...\n",
      "[M::mem_process_seqs] Processed 1668092 reads in 397.936 CPU sec, 33.217 real sec\n",
      "[M::mem_process_seqs] Processed 528178 reads in 131.544 CPU sec, 10.929 real sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa mem -t 12 /home/louie/Projects/Personal/GraphDB/Data/GWAS/Hsapiens_ref/ENSEMBL_GRCh38.p14/Homo_sapiens.GRCh38.dna_sm.primary_assembly.fa /home/louie/Projects/Personal/GraphDB/Data/GWAS/PRJNA1104541/SRR28808376/SRR28808376.fastq\n",
      "[main] Real time: 113.222 sec; CPU: 1325.151 sec\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[mpileup] 1 samples in 1 input files\n",
      "[mpileup] maximum number of reads per input file set to -d 250\n",
      "Note: none of --samples-file, --ploidy or --ploidy-file given, assuming all sites are diploid\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SRR28808376.fastq analysis complete | 3.7999088327089945 minutes\n",
      "\n",
      "\n",
      "~~~~~~~~~~\n",
      "(4) - Main file - SRR28808368.fastq\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 1682566 sequences (120000138 bp)...\n",
      "[M::process] read 1680954 sequences (120000047 bp)...\n",
      "[M::mem_process_seqs] Processed 1682566 reads in 406.668 CPU sec, 34.046 real sec\n",
      "[M::process] read 1680924 sequences (120000026 bp)...\n",
      "[M::mem_process_seqs] Processed 1680954 reads in 421.280 CPU sec, 35.175 real sec\n",
      "[M::process] read 1681850 sequences (120000081 bp)...\n",
      "[M::mem_process_seqs] Processed 1680924 reads in 421.163 CPU sec, 35.156 real sec\n",
      "[M::process] read 913485 sequences (65425445 bp)...\n",
      "[M::mem_process_seqs] Processed 1681850 reads in 430.103 CPU sec, 35.931 real sec\n",
      "[M::mem_process_seqs] Processed 913485 reads in 240.021 CPU sec, 20.025 real sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa mem -t 12 /home/louie/Projects/Personal/GraphDB/Data/GWAS/Hsapiens_ref/ENSEMBL_GRCh38.p14/Homo_sapiens.GRCh38.dna_sm.primary_assembly.fa /home/louie/Projects/Personal/GraphDB/Data/GWAS/PRJNA1104541/SRR28808368/SRR28808368.fastq\n",
      "[main] Real time: 162.842 sec; CPU: 1921.587 sec\n",
      "[bam_sort_core] merging from 2 files and 1 in-memory blocks...\n",
      "[mpileup] 1 samples in 1 input files\n",
      "[mpileup] maximum number of reads per input file set to -d 250\n",
      "Note: none of --samples-file, --ploidy or --ploidy-file given, assuming all sites are diploid\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SRR28808368.fastq analysis complete | 5.186338388919831 minutes\n",
      "\n",
      "\n",
      "~~~~~~~~~~\n",
      "(5) - Main file - SRR28808366.fastq\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 1679978 sequences (120000013 bp)...\n",
      "[M::process] read 1677978 sequences (120000115 bp)...\n",
      "[M::mem_process_seqs] Processed 1679978 reads in 422.884 CPU sec, 35.390 real sec\n",
      "[M::process] read 1510337 sequences (108084202 bp)...\n",
      "[M::mem_process_seqs] Processed 1677978 reads in 436.831 CPU sec, 36.452 real sec\n",
      "[M::mem_process_seqs] Processed 1510337 reads in 396.810 CPU sec, 33.171 real sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa mem -t 12 /home/louie/Projects/Personal/GraphDB/Data/GWAS/Hsapiens_ref/ENSEMBL_GRCh38.p14/Homo_sapiens.GRCh38.dna_sm.primary_assembly.fa /home/louie/Projects/Personal/GraphDB/Data/GWAS/PRJNA1104541/SRR28808366/SRR28808366.fastq\n",
      "[main] Real time: 109.522 sec; CPU: 1259.558 sec\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[mpileup] 1 samples in 1 input files\n",
      "[mpileup] maximum number of reads per input file set to -d 250\n",
      "Note: none of --samples-file, --ploidy or --ploidy-file given, assuming all sites are diploid\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SRR28808366.fastq analysis complete | 3.7316822608311973 minutes\n",
      "\n",
      "\n",
      "~~~~~~~~~~\n",
      "(6) - Main file - SRR28808389.fastq\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 1690904 sequences (120000092 bp)...\n",
      "[M::process] read 1688092 sequences (120000058 bp)...\n",
      "[M::mem_process_seqs] Processed 1690904 reads in 505.164 CPU sec, 42.301 real sec\n",
      "[M::process] read 1687396 sequences (120000122 bp)...\n",
      "[M::mem_process_seqs] Processed 1688092 reads in 507.423 CPU sec, 42.412 real sec\n",
      "[M::process] read 25372 sequences (1812298 bp)...\n",
      "[M::mem_process_seqs] Processed 1687396 reads in 512.816 CPU sec, 42.890 real sec\n",
      "[M::mem_process_seqs] Processed 25372 reads in 9.809 CPU sec, 0.811 real sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa mem -t 12 /home/louie/Projects/Personal/GraphDB/Data/GWAS/Hsapiens_ref/ENSEMBL_GRCh38.p14/Homo_sapiens.GRCh38.dna_sm.primary_assembly.fa /home/louie/Projects/Personal/GraphDB/Data/GWAS/PRJNA1104541/SRR28808389/SRR28808389.fastq\n",
      "[main] Real time: 131.934 sec; CPU: 1537.917 sec\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[mpileup] 1 samples in 1 input files\n",
      "[mpileup] maximum number of reads per input file set to -d 250\n",
      "Note: none of --samples-file, --ploidy or --ploidy-file given, assuming all sites are diploid\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SRR28808389.fastq analysis complete | 4.432262452443441 minutes\n",
      "\n",
      "\n",
      "~~~~~~~~~~\n",
      "(7) - Main file - SRR28808392.fastq\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 1645700 sequences (120000031 bp)...\n",
      "[M::process] read 1645516 sequences (120000106 bp)...\n",
      "[M::mem_process_seqs] Processed 1645700 reads in 472.500 CPU sec, 39.559 real sec\n",
      "[M::process] read 521781 sequences (38103286 bp)...\n",
      "[M::mem_process_seqs] Processed 1645516 reads in 472.105 CPU sec, 39.471 real sec\n",
      "[M::mem_process_seqs] Processed 521781 reads in 158.414 CPU sec, 13.176 real sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa mem -t 12 /home/louie/Projects/Personal/GraphDB/Data/GWAS/Hsapiens_ref/ENSEMBL_GRCh38.p14/Homo_sapiens.GRCh38.dna_sm.primary_assembly.fa /home/louie/Projects/Personal/GraphDB/Data/GWAS/PRJNA1104541/SRR28808392/SRR28808392.fastq\n",
      "[main] Real time: 96.223 sec; CPU: 1105.781 sec\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[mpileup] 1 samples in 1 input files\n",
      "[mpileup] maximum number of reads per input file set to -d 250\n",
      "Note: none of --samples-file, --ploidy or --ploidy-file given, assuming all sites are diploid\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SRR28808392.fastq analysis complete | 3.419305725892385 minutes\n",
      "\n",
      "\n",
      "~~~~~~~~~~\n",
      "(8) - Main file - SRR28808383.fastq\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 1673676 sequences (120000100 bp)...\n",
      "[M::process] read 1671248 sequences (120000058 bp)...\n",
      "[M::mem_process_seqs] Processed 1673676 reads in 396.268 CPU sec, 33.161 real sec\n",
      "[M::process] read 1673514 sequences (120000104 bp)...\n",
      "[M::mem_process_seqs] Processed 1671248 reads in 403.757 CPU sec, 33.699 real sec\n",
      "[M::process] read 727572 sequences (52348845 bp)...\n",
      "[M::mem_process_seqs] Processed 1673514 reads in 404.767 CPU sec, 33.806 real sec\n",
      "[M::mem_process_seqs] Processed 727572 reads in 180.946 CPU sec, 15.068 real sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa mem -t 12 /home/louie/Projects/Personal/GraphDB/Data/GWAS/Hsapiens_ref/ENSEMBL_GRCh38.p14/Homo_sapiens.GRCh38.dna_sm.primary_assembly.fa /home/louie/Projects/Personal/GraphDB/Data/GWAS/PRJNA1104541/SRR28808383/SRR28808383.fastq\n",
      "[main] Real time: 118.218 sec; CPU: 1388.007 sec\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[mpileup] 1 samples in 1 input files\n",
      "[mpileup] maximum number of reads per input file set to -d 250\n",
      "Note: none of --samples-file, --ploidy or --ploidy-file given, assuming all sites are diploid\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SRR28808383.fastq analysis complete | 4.257020231088003 minutes\n",
      "\n",
      "\n",
      "~~~~~~~~~~\n",
      "(9) - Main file - SRR28808381.fastq\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 1667042 sequences (120000126 bp)...\n",
      "[M::process] read 1666666 sequences (120000111 bp)...\n",
      "[M::mem_process_seqs] Processed 1667042 reads in 438.893 CPU sec, 36.726 real sec\n",
      "[M::process] read 650214 sequences (46925104 bp)...\n",
      "[M::mem_process_seqs] Processed 1666666 reads in 441.203 CPU sec, 36.866 real sec\n",
      "[M::mem_process_seqs] Processed 650214 reads in 178.633 CPU sec, 14.873 real sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa mem -t 12 /home/louie/Projects/Personal/GraphDB/Data/GWAS/Hsapiens_ref/ENSEMBL_GRCh38.p14/Homo_sapiens.GRCh38.dna_sm.primary_assembly.fa /home/louie/Projects/Personal/GraphDB/Data/GWAS/PRJNA1104541/SRR28808381/SRR28808381.fastq\n",
      "[main] Real time: 90.892 sec; CPU: 1061.023 sec\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[mpileup] 1 samples in 1 input files\n",
      "[mpileup] maximum number of reads per input file set to -d 250\n",
      "Note: none of --samples-file, --ploidy or --ploidy-file given, assuming all sites are diploid\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SRR28808381.fastq analysis complete | 3.260781459013621 minutes\n",
      "\n",
      "\n",
      "~~~~~~~~~~\n",
      "(10) - Main file - SRR28808372.fastq\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 1687048 sequences (120000107 bp)...\n",
      "[M::process] read 1683538 sequences (120000052 bp)...\n",
      "[M::mem_process_seqs] Processed 1687048 reads in 481.763 CPU sec, 40.335 real sec\n",
      "[M::process] read 1686222 sequences (120000120 bp)...\n",
      "[M::mem_process_seqs] Processed 1683538 reads in 489.346 CPU sec, 40.893 real sec\n",
      "[M::process] read 1684354 sequences (120000132 bp)...\n",
      "[M::mem_process_seqs] Processed 1686222 reads in 490.512 CPU sec, 40.990 real sec\n",
      "[M::process] read 396318 sequences (28377095 bp)...\n",
      "[M::mem_process_seqs] Processed 1684354 reads in 490.974 CPU sec, 41.077 real sec\n",
      "[M::mem_process_seqs] Processed 396318 reads in 124.028 CPU sec, 10.298 real sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa mem -t 12 /home/louie/Projects/Personal/GraphDB/Data/GWAS/Hsapiens_ref/ENSEMBL_GRCh38.p14/Homo_sapiens.GRCh38.dna_sm.primary_assembly.fa /home/louie/Projects/Personal/GraphDB/Data/GWAS/PRJNA1104541/SRR28808372/SRR28808372.fastq\n",
      "[main] Real time: 175.884 sec; CPU: 2078.868 sec\n",
      "[bam_sort_core] merging from 2 files and 1 in-memory blocks...\n",
      "[mpileup] 1 samples in 1 input files\n",
      "[mpileup] maximum number of reads per input file set to -d 250\n",
      "Note: none of --samples-file, --ploidy or --ploidy-file given, assuming all sites are diploid\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SRR28808372.fastq analysis complete | 7.501302234331766 minutes\n",
      "\n",
      "\n",
      "~~~~~~~~~~\n",
      "(11) - Main file - SRR28808388.fastq\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 1686380 sequences (120000053 bp)...\n",
      "[M::process] read 1684474 sequences (120000132 bp)...\n",
      "[M::mem_process_seqs] Processed 1686380 reads in 436.992 CPU sec, 36.587 real sec\n",
      "[M::process] read 1428805 sequences (101884526 bp)...\n",
      "[M::mem_process_seqs] Processed 1684474 reads in 447.836 CPU sec, 37.428 real sec\n",
      "[M::mem_process_seqs] Processed 1428805 reads in 376.833 CPU sec, 31.497 real sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa mem -t 12 /home/louie/Projects/Personal/GraphDB/Data/GWAS/Hsapiens_ref/ENSEMBL_GRCh38.p14/Homo_sapiens.GRCh38.dna_sm.primary_assembly.fa /home/louie/Projects/Personal/GraphDB/Data/GWAS/PRJNA1104541/SRR28808388/SRR28808388.fastq\n",
      "[main] Real time: 109.978 sec; CPU: 1264.328 sec\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[mpileup] 1 samples in 1 input files\n",
      "[mpileup] maximum number of reads per input file set to -d 250\n",
      "Note: none of --samples-file, --ploidy or --ploidy-file given, assuming all sites are diploid\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SRR28808388.fastq analysis complete | 3.90046413342158 minutes\n",
      "\n",
      "\n",
      "~~~~~~~~~~\n",
      "(12) - Main file - SRR28808378.fastq\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 1695934 sequences (120000097 bp)...\n",
      "[M::process] read 1694896 sequences (120000015 bp)...\n",
      "[M::mem_process_seqs] Processed 1695934 reads in 452.713 CPU sec, 37.906 real sec\n",
      "[M::process] read 759304 sequences (53912315 bp)...\n",
      "[M::mem_process_seqs] Processed 1694896 reads in 457.550 CPU sec, 38.231 real sec\n",
      "[M::mem_process_seqs] Processed 759304 reads in 214.994 CPU sec, 17.915 real sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa mem -t 12 /home/louie/Projects/Personal/GraphDB/Data/GWAS/Hsapiens_ref/ENSEMBL_GRCh38.p14/Homo_sapiens.GRCh38.dna_sm.primary_assembly.fa /home/louie/Projects/Personal/GraphDB/Data/GWAS/PRJNA1104541/SRR28808378/SRR28808378.fastq\n",
      "[main] Real time: 96.825 sec; CPU: 1127.660 sec\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[mpileup] 1 samples in 1 input files\n",
      "[mpileup] maximum number of reads per input file set to -d 250\n",
      "Note: none of --samples-file, --ploidy or --ploidy-file given, assuming all sites are diploid\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SRR28808378.fastq analysis complete | 3.3371157964070637 minutes\n",
      "\n",
      "\n",
      "~~~~~~~~~~\n",
      "(13) - Main file - SRR28808393.fastq\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 1668610 sequences (120000117 bp)...\n",
      "[M::process] read 1665650 sequences (120000144 bp)...\n",
      "[M::mem_process_seqs] Processed 1668610 reads in 452.654 CPU sec, 37.898 real sec\n",
      "[M::process] read 1667788 sequences (120000024 bp)...\n",
      "[M::mem_process_seqs] Processed 1665650 reads in 461.541 CPU sec, 38.530 real sec\n",
      "[M::process] read 716447 sequences (51710263 bp)...\n",
      "[M::mem_process_seqs] Processed 1667788 reads in 456.079 CPU sec, 38.102 real sec\n",
      "[M::mem_process_seqs] Processed 716447 reads in 203.909 CPU sec, 16.982 real sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa mem -t 12 /home/louie/Projects/Personal/GraphDB/Data/GWAS/Hsapiens_ref/ENSEMBL_GRCh38.p14/Homo_sapiens.GRCh38.dna_sm.primary_assembly.fa /home/louie/Projects/Personal/GraphDB/Data/GWAS/PRJNA1104541/SRR28808393/SRR28808393.fastq\n",
      "[main] Real time: 134.107 sec; CPU: 1576.504 sec\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[mpileup] 1 samples in 1 input files\n",
      "[mpileup] maximum number of reads per input file set to -d 250\n",
      "Note: none of --samples-file, --ploidy or --ploidy-file given, assuming all sites are diploid\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SRR28808393.fastq analysis complete | 5.906110076109568 minutes\n",
      "\n",
      "\n",
      "~~~~~~~~~~\n",
      "(14) - Main file - SRR28808364.fastq\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 1661356 sequences (120000076 bp)...\n",
      "[M::process] read 1659230 sequences (120000047 bp)...\n",
      "[M::mem_process_seqs] Processed 1661356 reads in 411.506 CPU sec, 34.400 real sec\n",
      "[M::process] read 1661288 sequences (120000022 bp)...\n",
      "[M::mem_process_seqs] Processed 1659230 reads in 420.938 CPU sec, 35.117 real sec\n",
      "[M::process] read 938286 sequences (67941941 bp)...\n",
      "[M::mem_process_seqs] Processed 1661288 reads in 415.768 CPU sec, 34.699 real sec\n",
      "[M::mem_process_seqs] Processed 938286 reads in 242.441 CPU sec, 20.209 real sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa mem -t 12 /home/louie/Projects/Personal/GraphDB/Data/GWAS/Hsapiens_ref/ENSEMBL_GRCh38.p14/Homo_sapiens.GRCh38.dna_sm.primary_assembly.fa /home/louie/Projects/Personal/GraphDB/Data/GWAS/PRJNA1104541/SRR28808364/SRR28808364.fastq\n",
      "[main] Real time: 133.283 sec; CPU: 1493.024 sec\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[mpileup] 1 samples in 1 input files\n",
      "[mpileup] maximum number of reads per input file set to -d 250\n",
      "Note: none of --samples-file, --ploidy or --ploidy-file given, assuming all sites are diploid\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SRR28808364.fastq analysis complete | 4.424142316977183 minutes\n",
      "\n",
      "\n",
      "~~~~~~~~~~\n",
      "(15) - Main file - SRR28808369.fastq\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 1677680 sequences (120000117 bp)...\n",
      "[M::process] read 1677464 sequences (120000072 bp)...\n",
      "[M::mem_process_seqs] Processed 1677680 reads in 413.646 CPU sec, 34.592 real sec\n",
      "[M::process] read 391297 sequences (28084361 bp)...\n",
      "[M::mem_process_seqs] Processed 1677464 reads in 422.187 CPU sec, 35.270 real sec\n",
      "[M::mem_process_seqs] Processed 391297 reads in 105.001 CPU sec, 8.704 real sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa mem -t 12 /home/louie/Projects/Personal/GraphDB/Data/GWAS/Hsapiens_ref/ENSEMBL_GRCh38.p14/Homo_sapiens.GRCh38.dna_sm.primary_assembly.fa /home/louie/Projects/Personal/GraphDB/Data/GWAS/PRJNA1104541/SRR28808369/SRR28808369.fastq\n",
      "[main] Real time: 81.094 sec; CPU: 943.074 sec\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[mpileup] 1 samples in 1 input files\n",
      "[mpileup] maximum number of reads per input file set to -d 250\n",
      "Note: none of --samples-file, --ploidy or --ploidy-file given, assuming all sites are diploid\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SRR28808369.fastq analysis complete | 2.807760246594747 minutes\n",
      "\n",
      "\n",
      "~~~~~~~~~~\n",
      "(16) - Main file - SRR28808380.fastq\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 1678334 sequences (120000104 bp)...\n",
      "[M::process] read 1677568 sequences (120000027 bp)...\n",
      "[M::mem_process_seqs] Processed 1678334 reads in 403.601 CPU sec, 33.768 real sec\n",
      "[M::process] read 980686 sequences (70291721 bp)...\n",
      "[M::mem_process_seqs] Processed 1677568 reads in 405.812 CPU sec, 33.871 real sec\n",
      "[M::mem_process_seqs] Processed 980686 reads in 241.039 CPU sec, 20.093 real sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa mem -t 12 /home/louie/Projects/Personal/GraphDB/Data/GWAS/Hsapiens_ref/ENSEMBL_GRCh38.p14/Homo_sapiens.GRCh38.dna_sm.primary_assembly.fa /home/louie/Projects/Personal/GraphDB/Data/GWAS/PRJNA1104541/SRR28808380/SRR28808380.fastq\n",
      "[main] Real time: 90.180 sec; CPU: 1052.767 sec\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[mpileup] 1 samples in 1 input files\n",
      "[mpileup] maximum number of reads per input file set to -d 250\n",
      "Note: none of --samples-file, --ploidy or --ploidy-file given, assuming all sites are diploid\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SRR28808380.fastq analysis complete | 3.3272207816441854 minutes\n",
      "\n",
      "\n",
      "~~~~~~~~~~\n",
      "(17) - Main file - SRR28808374.fastq\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 1676020 sequences (120000047 bp)...\n",
      "[M::process] read 1673362 sequences (120000100 bp)...\n",
      "[M::mem_process_seqs] Processed 1676020 reads in 425.930 CPU sec, 35.618 real sec\n",
      "[M::process] read 1676228 sequences (120000115 bp)...\n",
      "[M::mem_process_seqs] Processed 1673362 reads in 430.498 CPU sec, 35.943 real sec\n",
      "[M::process] read 919876 sequences (66071616 bp)...\n",
      "[M::mem_process_seqs] Processed 1676228 reads in 431.919 CPU sec, 36.052 real sec\n",
      "[M::mem_process_seqs] Processed 919876 reads in 240.813 CPU sec, 20.065 real sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa mem -t 12 /home/louie/Projects/Personal/GraphDB/Data/GWAS/Hsapiens_ref/ENSEMBL_GRCh38.p14/Homo_sapiens.GRCh38.dna_sm.primary_assembly.fa /home/louie/Projects/Personal/GraphDB/Data/GWAS/PRJNA1104541/SRR28808374/SRR28808374.fastq\n",
      "[main] Real time: 130.668 sec; CPU: 1531.523 sec\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[mpileup] 1 samples in 1 input files\n",
      "[mpileup] maximum number of reads per input file set to -d 250\n",
      "Note: none of --samples-file, --ploidy or --ploidy-file given, assuming all sites are diploid\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SRR28808374.fastq analysis complete | 4.2315562009811405 minutes\n",
      "\n",
      "\n",
      "~~~~~~~~~~\n",
      "(18) - Main file - SRR28808373.fastq\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 1675184 sequences (120000093 bp)...\n",
      "[M::process] read 1672474 sequences (120000048 bp)...\n",
      "[M::mem_process_seqs] Processed 1675184 reads in 360.565 CPU sec, 30.176 real sec\n",
      "[M::process] read 1675208 sequences (120000148 bp)...\n",
      "[M::mem_process_seqs] Processed 1672474 reads in 371.858 CPU sec, 31.021 real sec\n",
      "[M::process] read 972919 sequences (69909899 bp)...\n",
      "[M::mem_process_seqs] Processed 1675208 reads in 367.145 CPU sec, 30.630 real sec\n",
      "[M::mem_process_seqs] Processed 972919 reads in 221.849 CPU sec, 18.486 real sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa mem -t 12 /home/louie/Projects/Personal/GraphDB/Data/GWAS/Hsapiens_ref/ENSEMBL_GRCh38.p14/Homo_sapiens.GRCh38.dna_sm.primary_assembly.fa /home/louie/Projects/Personal/GraphDB/Data/GWAS/PRJNA1104541/SRR28808373/SRR28808373.fastq\n",
      "[main] Real time: 113.022 sec; CPU: 1323.758 sec\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[mpileup] 1 samples in 1 input files\n",
      "[mpileup] maximum number of reads per input file set to -d 250\n",
      "Note: none of --samples-file, --ploidy or --ploidy-file given, assuming all sites are diploid\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SRR28808373.fastq analysis complete | 3.942350419362386 minutes\n",
      "\n",
      "\n",
      "~~~~~~~~~~\n",
      "(19) - Main file - SRR28808385.fastq\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 1698476 sequences (120000102 bp)...\n",
      "[M::process] read 1696022 sequences (120000039 bp)...\n",
      "[M::mem_process_seqs] Processed 1698476 reads in 453.054 CPU sec, 47.898 real sec\n",
      "[M::process] read 1274413 sequences (90284724 bp)...\n",
      "[M::mem_process_seqs] Processed 1696022 reads in 462.856 CPU sec, 41.626 real sec\n",
      "[M::mem_process_seqs] Processed 1274413 reads in 349.403 CPU sec, 29.174 real sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa mem -t 12 /home/louie/Projects/Personal/GraphDB/Data/GWAS/Hsapiens_ref/ENSEMBL_GRCh38.p14/Homo_sapiens.GRCh38.dna_sm.primary_assembly.fa /home/louie/Projects/Personal/GraphDB/Data/GWAS/PRJNA1104541/SRR28808385/SRR28808385.fastq\n",
      "[main] Real time: 121.173 sec; CPU: 1267.670 sec\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[mpileup] 1 samples in 1 input files\n",
      "[mpileup] maximum number of reads per input file set to -d 250\n",
      "Note: none of --samples-file, --ploidy or --ploidy-file given, assuming all sites are diploid\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SRR28808385.fastq analysis complete | 4.013261131445566 minutes\n",
      "\n",
      "\n",
      "~~~~~~~~~~\n",
      "(20) - Main file - SRR28808391.fastq\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 1677100 sequences (120000002 bp)...\n",
      "[M::process] read 1676848 sequences (120000065 bp)...\n",
      "[M::mem_process_seqs] Processed 1677100 reads in 409.951 CPU sec, 34.272 real sec\n",
      "[M::process] read 431237 sequences (30987494 bp)...\n",
      "[M::mem_process_seqs] Processed 1676848 reads in 412.619 CPU sec, 34.455 real sec\n",
      "[M::mem_process_seqs] Processed 431237 reads in 112.853 CPU sec, 9.362 real sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa mem -t 12 /home/louie/Projects/Personal/GraphDB/Data/GWAS/Hsapiens_ref/ENSEMBL_GRCh38.p14/Homo_sapiens.GRCh38.dna_sm.primary_assembly.fa /home/louie/Projects/Personal/GraphDB/Data/GWAS/PRJNA1104541/SRR28808391/SRR28808391.fastq\n",
      "[main] Real time: 80.530 sec; CPU: 937.670 sec\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[mpileup] 1 samples in 1 input files\n",
      "[mpileup] maximum number of reads per input file set to -d 250\n",
      "Note: none of --samples-file, --ploidy or --ploidy-file given, assuming all sites are diploid\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SRR28808391.fastq analysis complete | 2.959898002942403 minutes\n",
      "\n",
      "\n",
      "~~~~~~~~~~\n",
      "(21) - Main file - SRR28808379.fastq\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 1665304 sequences (120000017 bp)...\n",
      "[M::process] read 1665090 sequences (120000027 bp)...\n",
      "[M::mem_process_seqs] Processed 1665304 reads in 413.536 CPU sec, 34.574 real sec\n",
      "[M::process] read 130022 sequences (9392028 bp)...\n",
      "[M::mem_process_seqs] Processed 1665090 reads in 421.463 CPU sec, 35.189 real sec\n",
      "[M::mem_process_seqs] Processed 130022 reads in 40.387 CPU sec, 3.297 real sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa mem -t 12 /home/louie/Projects/Personal/GraphDB/Data/GWAS/Hsapiens_ref/ENSEMBL_GRCh38.p14/Homo_sapiens.GRCh38.dna_sm.primary_assembly.fa /home/louie/Projects/Personal/GraphDB/Data/GWAS/PRJNA1104541/SRR28808379/SRR28808379.fastq\n",
      "[main] Real time: 75.512 sec; CPU: 877.519 sec\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[mpileup] 1 samples in 1 input files\n",
      "[mpileup] maximum number of reads per input file set to -d 250\n",
      "Note: none of --samples-file, --ploidy or --ploidy-file given, assuming all sites are diploid\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SRR28808379.fastq analysis complete | 2.7009853800137837 minutes\n",
      "\n",
      "\n",
      "~~~~~~~~~~\n",
      "(22) - Main file - SRR28808377.fastq\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 1705844 sequences (120000039 bp)...\n",
      "[M::process] read 1706074 sequences (120000009 bp)...\n",
      "[M::mem_process_seqs] Processed 1705844 reads in 411.392 CPU sec, 34.427 real sec\n",
      "[M::process] read 681143 sequences (47988824 bp)...\n",
      "[M::mem_process_seqs] Processed 1706074 reads in 414.560 CPU sec, 34.621 real sec\n",
      "[M::mem_process_seqs] Processed 681143 reads in 174.338 CPU sec, 14.505 real sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa mem -t 12 /home/louie/Projects/Personal/GraphDB/Data/GWAS/Hsapiens_ref/ENSEMBL_GRCh38.p14/Homo_sapiens.GRCh38.dna_sm.primary_assembly.fa /home/louie/Projects/Personal/GraphDB/Data/GWAS/PRJNA1104541/SRR28808377/SRR28808377.fastq\n",
      "[main] Real time: 87.483 sec; CPU: 1002.630 sec\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[mpileup] 1 samples in 1 input files\n",
      "[mpileup] maximum number of reads per input file set to -d 250\n",
      "Note: none of --samples-file, --ploidy or --ploidy-file given, assuming all sites are diploid\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SRR28808377.fastq analysis complete | 3.0734646598498028 minutes\n",
      "\n",
      "\n",
      "~~~~~~~~~~\n",
      "(23) - Main file - SRR28808370.fastq\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 1685646 sequences (120000016 bp)...\n",
      "[M::process] read 1682444 sequences (120000055 bp)...\n",
      "[M::mem_process_seqs] Processed 1685646 reads in 356.812 CPU sec, 29.852 real sec\n",
      "[M::process] read 1685282 sequences (120000039 bp)...\n",
      "[M::mem_process_seqs] Processed 1682444 reads in 366.769 CPU sec, 30.585 real sec\n",
      "[M::process] read 857463 sequences (61284534 bp)...\n",
      "[M::mem_process_seqs] Processed 1685282 reads in 361.028 CPU sec, 30.125 real sec\n",
      "[M::mem_process_seqs] Processed 857463 reads in 193.913 CPU sec, 16.149 real sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa mem -t 12 /home/louie/Projects/Personal/GraphDB/Data/GWAS/Hsapiens_ref/ENSEMBL_GRCh38.p14/Homo_sapiens.GRCh38.dna_sm.primary_assembly.fa /home/louie/Projects/Personal/GraphDB/Data/GWAS/PRJNA1104541/SRR28808370/SRR28808370.fastq\n",
      "[main] Real time: 109.208 sec; CPU: 1280.804 sec\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[mpileup] 1 samples in 1 input files\n",
      "[mpileup] maximum number of reads per input file set to -d 250\n",
      "Note: none of --samples-file, --ploidy or --ploidy-file given, assuming all sites are diploid\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SRR28808370.fastq analysis complete | 3.488254185517629 minutes\n",
      "\n",
      "\n",
      "~~~~~~~~~~\n",
      "(24) - Main file - SRR28808367.fastq\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 1699218 sequences (120000011 bp)...\n",
      "[M::process] read 1697690 sequences (120000116 bp)...\n",
      "[M::mem_process_seqs] Processed 1699218 reads in 433.175 CPU sec, 36.217 real sec\n",
      "[M::process] read 1694302 sequences (120000076 bp)...\n",
      "[M::mem_process_seqs] Processed 1697690 reads in 436.949 CPU sec, 36.475 real sec\n",
      "[M::process] read 1699228 sequences (120000020 bp)...\n",
      "[M::mem_process_seqs] Processed 1694302 reads in 439.900 CPU sec, 36.704 real sec\n",
      "[M::process] read 1692898 sequences (120000049 bp)...\n",
      "[M::mem_process_seqs] Processed 1699228 reads in 434.651 CPU sec, 36.261 real sec\n",
      "[M::process] read 215206 sequences (15315721 bp)...\n",
      "[M::mem_process_seqs] Processed 1692898 reads in 441.068 CPU sec, 36.827 real sec\n",
      "[M::mem_process_seqs] Processed 215206 reads in 62.907 CPU sec, 5.182 real sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa mem -t 12 /home/louie/Projects/Personal/GraphDB/Data/GWAS/Hsapiens_ref/ENSEMBL_GRCh38.p14/Homo_sapiens.GRCh38.dna_sm.primary_assembly.fa /home/louie/Projects/Personal/GraphDB/Data/GWAS/PRJNA1104541/SRR28808367/SRR28808367.fastq\n",
      "[main] Real time: 190.038 sec; CPU: 2250.800 sec\n",
      "[bam_sort_core] merging from 2 files and 1 in-memory blocks...\n",
      "[mpileup] 1 samples in 1 input files\n",
      "[mpileup] maximum number of reads per input file set to -d 250\n",
      "Note: none of --samples-file, --ploidy or --ploidy-file given, assuming all sites are diploid\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SRR28808367.fastq analysis complete | 7.852405496438345 minutes\n",
      "\n",
      "\n",
      "~~~~~~~~~~\n",
      "(25) - Main file - SRR28808365.fastq\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 1683120 sequences (120000061 bp)...\n",
      "[M::process] read 1679386 sequences (120000063 bp)...\n",
      "[M::mem_process_seqs] Processed 1683120 reads in 473.790 CPU sec, 39.630 real sec\n",
      "[M::process] read 1682700 sequences (120000051 bp)...\n",
      "[M::mem_process_seqs] Processed 1679386 reads in 479.814 CPU sec, 40.075 real sec\n",
      "[M::process] read 1679556 sequences (120000019 bp)...\n",
      "[M::mem_process_seqs] Processed 1682700 reads in 475.984 CPU sec, 39.725 real sec\n",
      "[M::process] read 245703 sequences (17621820 bp)...\n",
      "[M::mem_process_seqs] Processed 1679556 reads in 480.108 CPU sec, 40.084 real sec\n",
      "[M::mem_process_seqs] Processed 245703 reads in 76.985 CPU sec, 6.357 real sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa mem -t 12 /home/louie/Projects/Personal/GraphDB/Data/GWAS/Hsapiens_ref/ENSEMBL_GRCh38.p14/Homo_sapiens.GRCh38.dna_sm.primary_assembly.fa /home/louie/Projects/Personal/GraphDB/Data/GWAS/PRJNA1104541/SRR28808365/SRR28808365.fastq\n",
      "[main] Real time: 170.198 sec; CPU: 1989.085 sec\n",
      "[bam_sort_core] merging from 2 files and 1 in-memory blocks...\n",
      "[mpileup] 1 samples in 1 input files\n",
      "[mpileup] maximum number of reads per input file set to -d 250\n",
      "Note: none of --samples-file, --ploidy or --ploidy-file given, assuming all sites are diploid\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SRR28808365.fastq analysis complete | 5.890706038475036 minutes\n",
      "\n",
      "\n",
      "~~~~~~~~~~\n",
      "(26) - Main file - SRR28808384.fastq\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 1685016 sequences (120000077 bp)...\n",
      "[M::process] read 1682544 sequences (120000119 bp)...\n",
      "[M::mem_process_seqs] Processed 1685016 reads in 429.300 CPU sec, 35.885 real sec\n",
      "[M::process] read 1546971 sequences (110441658 bp)...\n",
      "[M::mem_process_seqs] Processed 1682544 reads in 438.616 CPU sec, 36.602 real sec\n",
      "[M::mem_process_seqs] Processed 1546971 reads in 404.755 CPU sec, 33.817 real sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa mem -t 12 /home/louie/Projects/Personal/GraphDB/Data/GWAS/Hsapiens_ref/ENSEMBL_GRCh38.p14/Homo_sapiens.GRCh38.dna_sm.primary_assembly.fa /home/louie/Projects/Personal/GraphDB/Data/GWAS/PRJNA1104541/SRR28808384/SRR28808384.fastq\n",
      "[main] Real time: 109.394 sec; CPU: 1275.160 sec\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[mpileup] 1 samples in 1 input files\n",
      "[mpileup] maximum number of reads per input file set to -d 250\n",
      "Note: none of --samples-file, --ploidy or --ploidy-file given, assuming all sites are diploid\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SRR28808384.fastq analysis complete | 3.5652425289154053 minutes\n",
      "\n",
      "\n",
      "~~~~~~~~~~\n",
      "(27) - Main file - SRR28808371.fastq\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 1675178 sequences (120000000 bp)...\n",
      "[M::process] read 1673842 sequences (120000076 bp)...\n",
      "[M::mem_process_seqs] Processed 1675178 reads in 493.065 CPU sec, 41.245 real sec\n",
      "[M::process] read 1672292 sequences (120000095 bp)...\n",
      "[M::mem_process_seqs] Processed 1673842 reads in 498.925 CPU sec, 41.662 real sec\n",
      "[M::process] read 1674858 sequences (120000138 bp)...\n",
      "[M::mem_process_seqs] Processed 1672292 reads in 502.783 CPU sec, 41.956 real sec\n",
      "[M::process] read 1394469 sequences (100242746 bp)...\n",
      "[M::mem_process_seqs] Processed 1674858 reads in 497.093 CPU sec, 41.481 real sec\n",
      "[M::mem_process_seqs] Processed 1394469 reads in 420.923 CPU sec, 35.142 real sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa mem -t 12 /home/louie/Projects/Personal/GraphDB/Data/GWAS/Hsapiens_ref/ENSEMBL_GRCh38.p14/Homo_sapiens.GRCh38.dna_sm.primary_assembly.fa /home/louie/Projects/Personal/GraphDB/Data/GWAS/PRJNA1104541/SRR28808371/SRR28808371.fastq\n",
      "[main] Real time: 203.909 sec; CPU: 2415.087 sec\n",
      "[bam_sort_core] merging from 2 files and 1 in-memory blocks...\n",
      "[mpileup] 1 samples in 1 input files\n",
      "[mpileup] maximum number of reads per input file set to -d 250\n",
      "Note: none of --samples-file, --ploidy or --ploidy-file given, assuming all sites are diploid\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SRR28808371.fastq analysis complete | 6.767438018321991 minutes\n",
      "\n",
      "\n",
      "~~~~~~~~~~\n",
      "(28) - Main file - SRR28808375.fastq\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 1696894 sequences (120000134 bp)...\n",
      "[M::process] read 1696102 sequences (120000022 bp)...\n",
      "[M::mem_process_seqs] Processed 1696894 reads in 425.793 CPU sec, 35.608 real sec\n",
      "[M::process] read 1692034 sequences (120000033 bp)...\n",
      "[M::mem_process_seqs] Processed 1696102 reads in 431.049 CPU sec, 35.965 real sec\n",
      "[M::process] read 1697270 sequences (120000103 bp)...\n",
      "[M::mem_process_seqs] Processed 1692034 reads in 433.948 CPU sec, 36.215 real sec\n",
      "[M::process] read 1693926 sequences (120000139 bp)...\n",
      "[M::mem_process_seqs] Processed 1697270 reads in 427.889 CPU sec, 35.707 real sec\n",
      "[M::process] read 664655 sequences (47263702 bp)...\n",
      "[M::mem_process_seqs] Processed 1693926 reads in 432.043 CPU sec, 36.075 real sec\n",
      "[M::mem_process_seqs] Processed 664655 reads in 174.534 CPU sec, 14.527 real sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa mem -t 12 /home/louie/Projects/Personal/GraphDB/Data/GWAS/Hsapiens_ref/ENSEMBL_GRCh38.p14/Homo_sapiens.GRCh38.dna_sm.primary_assembly.fa /home/louie/Projects/Personal/GraphDB/Data/GWAS/PRJNA1104541/SRR28808375/SRR28808375.fastq\n",
      "[main] Real time: 196.770 sec; CPU: 2327.482 sec\n",
      "[bam_sort_core] merging from 2 files and 1 in-memory blocks...\n",
      "[mpileup] 1 samples in 1 input files\n",
      "[mpileup] maximum number of reads per input file set to -d 250\n",
      "Note: none of --samples-file, --ploidy or --ploidy-file given, assuming all sites are diploid\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SRR28808375.fastq analysis complete | 7.125429578622183 minutes\n",
      "\n",
      "\n",
      "~~~~~~~~~~\n",
      "(29) - Main file - SRR28808382.fastq\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 1674202 sequences (120000012 bp)...\n",
      "[M::process] read 1672530 sequences (120000088 bp)...\n",
      "[M::mem_process_seqs] Processed 1674202 reads in 445.400 CPU sec, 37.240 real sec\n",
      "[M::process] read 1246586 sequences (89519403 bp)...\n",
      "[M::mem_process_seqs] Processed 1672530 reads in 452.796 CPU sec, 37.830 real sec\n",
      "[M::mem_process_seqs] Processed 1246586 reads in 338.372 CPU sec, 28.232 real sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa mem -t 12 /home/louie/Projects/Personal/GraphDB/Data/GWAS/Hsapiens_ref/ENSEMBL_GRCh38.p14/Homo_sapiens.GRCh38.dna_sm.primary_assembly.fa /home/louie/Projects/Personal/GraphDB/Data/GWAS/PRJNA1104541/SRR28808382/SRR28808382.fastq\n",
      "[main] Real time: 106.361 sec; CPU: 1238.949 sec\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[mpileup] 1 samples in 1 input files\n",
      "[mpileup] maximum number of reads per input file set to -d 250\n",
      "Note: none of --samples-file, --ploidy or --ploidy-file given, assuming all sites are diploid\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SRR28808382.fastq analysis complete | 3.6728576461474103 minutes\n",
      "\n",
      "\n",
      "~~~~~~~~~~\n",
      "(30) - Main file - SRR28808387.fastq\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 1659840 sequences (120000049 bp)...\n",
      "[M::process] read 1658732 sequences (120000146 bp)...\n",
      "[M::mem_process_seqs] Processed 1659840 reads in 488.968 CPU sec, 40.895 real sec\n",
      "[M::process] read 1459568 sequences (105639900 bp)...\n",
      "[M::mem_process_seqs] Processed 1658732 reads in 496.526 CPU sec, 41.431 real sec\n",
      "[M::mem_process_seqs] Processed 1459568 reads in 436.774 CPU sec, 36.482 real sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa mem -t 12 /home/louie/Projects/Personal/GraphDB/Data/GWAS/Hsapiens_ref/ENSEMBL_GRCh38.p14/Homo_sapiens.GRCh38.dna_sm.primary_assembly.fa /home/louie/Projects/Personal/GraphDB/Data/GWAS/PRJNA1104541/SRR28808387/SRR28808387.fastq\n",
      "[main] Real time: 121.168 sec; CPU: 1424.569 sec\n",
      "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
      "[mpileup] 1 samples in 1 input files\n",
      "[mpileup] maximum number of reads per input file set to -d 250\n",
      "Note: none of --samples-file, --ploidy or --ploidy-file given, assuming all sites are diploid\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SRR28808387.fastq analysis complete | 4.088012055555979 minutes\n"
     ]
    }
   ],
   "source": [
    "maindir   = datadir+\"/GWAS/PRJNA1104541/\"\n",
    "snpeff    = \"/home/louie/Documents/Software/snpEff/snpEff.jar\"\n",
    "\n",
    "count = 0\n",
    "for i in os.listdir(maindir):\n",
    "    start = time.time()\n",
    "    if \"fastq\" in i and \".gz\" not in i:\n",
    "        count += 1\n",
    "        stem = i.split(\".fastq\")[0]\n",
    "        loc_short = f'{maindir}{stem}/{stem}'\n",
    "        print(f'{\"\\n\"*2}{\"~\"*10}\\n({count}) - Main file - {i}')\n",
    "        # Make directory\n",
    "        if os.path.isdir(f'{maindir}{stem}') == False:\n",
    "            os.system(f'mkdir {maindir}{stem}')\n",
    "        os.system(f'cp {maindir}{i} {maindir}{stem}')\n",
    "        # Align reads to reference (single-end)\n",
    "        os.system(f'\\nbwa mem -t 12 {reference} {loc_short}.fastq > {loc_short}_aligned.sam')\n",
    "        # Convert SAM to BAM\n",
    "        os.system(f'\\nsamtools view -Sb {loc_short}_aligned.sam | samtools sort -o {loc_short}_aligned.bam')\n",
    "        # Pileup generation\n",
    "        os.system(f'\\nbcftools mpileup -f {reference} {loc_short}_aligned.bam > {loc_short}.bcf')\n",
    "        # Call variants\n",
    "        os.system(f'\\nbcftools call -mv -Ov -o {loc_short}.vcf {loc_short}.bcf')\n",
    "        # Annotate VCF\n",
    "        os.system(f'\\njava -Xmx4g -jar {snpeff} GRCh38.p14 {loc_short}.vcf > {loc_short}_annotated.vcf')\n",
    "        # Clean intermediary files\n",
    "        os.system(f'rm {loc_short}.bcf {loc_short}_aligned.bam {loc_short}_aligned.sam {loc_short}.fastq {loc_short}.vcf')\n",
    "        print(f'{i} analysis complete | {round((time.time()-start)/60, 3)} minutes')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "92b007eb-cf2e-4d5c-b18c-30580992a26e",
   "metadata": {},
   "outputs": [],
   "source": [
    "import time\n",
    "import os\n",
    "import pandas as pd\n",
    "from cyvcf2 import VCF\n",
    "\n",
    "#Data path\n",
    "datadir = os.path.abspath(\"../../Data/\")\n",
    "\n",
    "#maindir   = datadir+\"/GWAS/PRJNA1104541/\"\n",
    "#snpeff    = \"/home/louie/Documents/Software/snpEff/snpEff.jar\"\n",
    "\n",
    "def IndexRefGenome(reference):\n",
    "    start = time.time()\n",
    "    # Index the reference genome with BWA\n",
    "    os.system(f'bwa index {reference}')\n",
    "    print(f'Reference index complete | {(time.time()-start)/60} minutes')\n",
    "\n",
    "def GenerateVCF(maindir, snpeff):\n",
    "    count = 0\n",
    "    for i in os.listdir(maindir):\n",
    "        start = time.time()\n",
    "        if \"fastq\" in i and \".gz\" not in i:\n",
    "            count += 1\n",
    "            stem = i.split(\".fastq\")[0]\n",
    "            loc_short = f'{maindir}{stem}/{stem}'\n",
    "            print(f'{\"\\n\"*2}{\"~\"*10}\\n({count}) - Main file - {i}')\n",
    "            # Make directory\n",
    "            if os.path.isdir(f'{maindir}{stem}') == False:\n",
    "                os.system(f'mkdir {maindir}{stem}')\n",
    "            os.system(f'cp {maindir}{i} {maindir}{stem}')\n",
    "            # Align reads to reference (single-end)\n",
    "            os.system(f'\\nbwa mem -t 12 {reference} {loc_short}.fastq > {loc_short}_aligned.sam')\n",
    "            # Convert SAM to BAM\n",
    "            os.system(f'\\nsamtools view -Sb {loc_short}_aligned.sam | samtools sort -o {loc_short}_aligned.bam')\n",
    "            # Pileup generation\n",
    "            os.system(f'\\nbcftools mpileup -f {reference} {loc_short}_aligned.bam > {loc_short}.bcf')\n",
    "            # Call variants\n",
    "            os.system(f'\\nbcftools call -mv -Ov -o {loc_short}.vcf {loc_short}.bcf')\n",
    "            # Annotate VCF\n",
    "            os.system(f'\\njava -Xmx4g -jar {snpeff} GRCh38.p14 {loc_short}.vcf > {loc_short}_annotated.vcf')\n",
    "            # Clean intermediary files\n",
    "            os.system(f'rm {loc_short}.bcf {loc_short}_aligned.bam {loc_short}_aligned.sam {loc_short}.fastq {loc_short}.vcf')\n",
    "            print(f'{i} analysis complete | {round((time.time()-start)/60, 3)} minutes')\n",
    "\n",
    "def ParseVCF(maindir):\n",
    "    #Set default df\n",
    "    df        = pd.DataFrame(columns = [\"Chr\", \"Pos\",\"Ref\", \"Alt\",\"Gene\",\"Effect\",\"Impact\",\"Transcript\",\"ProteinChange\"])\n",
    "\n",
    "    #Iterate through VCF files in directory, return df with \"High\" impact varients\n",
    "    start = time.time()\n",
    "    for path, subdirs, files in os.walk(maindir):\n",
    "        for name in files:\n",
    "            if \"annotated.vcf\" in name:\n",
    "                vcf_file = os.path.join(path, name)\n",
    "                print(vcf_file)\n",
    "                records = []\n",
    "                for variant in VCF(vcf_file):\n",
    "                    ann_field = variant.INFO.get(\"ANN\")\n",
    "                    for ann in ann_field.split(\",\"):\n",
    "                        fields = ann.split(\"|\")\n",
    "                        records.append({\n",
    "                            \"Chr\": variant.CHROM,\n",
    "                            \"Pos\": variant.POS,\n",
    "                            \"Ref\": variant.REF,\n",
    "                            \"Alt\": \",\".join(variant.ALT),\n",
    "                            \"Gene\": fields[3],\n",
    "                            \"Effect\": fields[1],\n",
    "                            \"Impact\": fields[2],\n",
    "                            \"Transcript\": fields[6],\n",
    "                            \"ProteinChange\": fields[10]\n",
    "                        })\n",
    "                tmpdf = pd.DataFrame(records)\n",
    "                df = pd.concat([df, tmpdf.loc[tmpdf[\"Impact\"] == \"HIGH\"]])\n",
    "                \n",
    "    print(f'DF generation complete | {round((time.time()-start)/60, 3)} minutes')\n",
    "    \n",
    "    return df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "9babfe60-0e5d-4365-950f-abc0dfeada6b",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/louie/Projects/Personal/GraphDB/Data/GWAS/PRJNA1104541/SRR28808368/SRR28808368_annotated.vcf\n",
      "/home/louie/Projects/Personal/GraphDB/Data/GWAS/PRJNA1104541/SRR28808377/SRR28808377_annotated.vcf\n",
      "/home/louie/Projects/Personal/GraphDB/Data/GWAS/PRJNA1104541/SRR28808385/SRR28808385_annotated.vcf\n",
      "/home/louie/Projects/Personal/GraphDB/Data/GWAS/PRJNA1104541/SRR28808370/SRR28808370_annotated.vcf\n",
      "/home/louie/Projects/Personal/GraphDB/Data/GWAS/PRJNA1104541/SRR28808381/SRR28808381_annotated.vcf\n",
      "/home/louie/Projects/Personal/GraphDB/Data/GWAS/PRJNA1104541/SRR28808392/SRR28808392_annotated.vcf\n",
      "/home/louie/Projects/Personal/GraphDB/Data/GWAS/PRJNA1104541/SRR28808364/SRR28808364_annotated.vcf\n",
      "/home/louie/Projects/Personal/GraphDB/Data/GWAS/PRJNA1104541/SRR28808379/SRR28808379_annotated.vcf\n",
      "/home/louie/Projects/Personal/GraphDB/Data/GWAS/PRJNA1104541/SRR28808388/SRR28808388_annotated.vcf\n",
      "/home/louie/Projects/Personal/GraphDB/Data/GWAS/PRJNA1104541/SRR28808371/SRR28808371_annotated.vcf\n",
      "/home/louie/Projects/Personal/GraphDB/Data/GWAS/PRJNA1104541/SRR28808387/SRR28808387_annotated.vcf\n",
      "/home/louie/Projects/Personal/GraphDB/Data/GWAS/PRJNA1104541/SRR28808380/SRR28808380_annotated.vcf\n",
      "/home/louie/Projects/Personal/GraphDB/Data/GWAS/PRJNA1104541/SRR28808393/SRR28808393_annotated.vcf\n",
      "/home/louie/Projects/Personal/GraphDB/Data/GWAS/PRJNA1104541/SRR28808367/SRR28808367_annotated.vcf\n",
      "/home/louie/Projects/Personal/GraphDB/Data/GWAS/PRJNA1104541/SRR28808365/SRR28808365_annotated.vcf\n",
      "/home/louie/Projects/Personal/GraphDB/Data/GWAS/PRJNA1104541/SRR28808374/SRR28808374_annotated.vcf\n",
      "/home/louie/Projects/Personal/GraphDB/Data/GWAS/PRJNA1104541/SRR28808378/SRR28808378_annotated.vcf\n",
      "/home/louie/Projects/Personal/GraphDB/Data/GWAS/PRJNA1104541/SRR28808386/SRR28808386_annotated.vcf\n",
      "/home/louie/Projects/Personal/GraphDB/Data/GWAS/PRJNA1104541/SRR28808375/SRR28808375_annotated.vcf\n",
      "/home/louie/Projects/Personal/GraphDB/Data/GWAS/PRJNA1104541/SRR28808383/SRR28808383_annotated.vcf\n",
      "/home/louie/Projects/Personal/GraphDB/Data/GWAS/PRJNA1104541/SRR28808373/SRR28808373_annotated.vcf\n",
      "/home/louie/Projects/Personal/GraphDB/Data/GWAS/PRJNA1104541/SRR28808389/SRR28808389_annotated.vcf\n",
      "/home/louie/Projects/Personal/GraphDB/Data/GWAS/PRJNA1104541/SRR28808366/SRR28808366_annotated.vcf\n",
      "/home/louie/Projects/Personal/GraphDB/Data/GWAS/PRJNA1104541/SRR28808384/SRR28808384_annotated.vcf\n",
      "/home/louie/Projects/Personal/GraphDB/Data/GWAS/PRJNA1104541/SRR28808376/SRR28808376_annotated.vcf\n",
      "/home/louie/Projects/Personal/GraphDB/Data/GWAS/PRJNA1104541/SRR28808390/SRR28808390_annotated.vcf\n",
      "/home/louie/Projects/Personal/GraphDB/Data/GWAS/PRJNA1104541/SRR28808382/SRR28808382_annotated.vcf\n",
      "/home/louie/Projects/Personal/GraphDB/Data/GWAS/PRJNA1104541/SRR28808369/SRR28808369_annotated.vcf\n",
      "/home/louie/Projects/Personal/GraphDB/Data/GWAS/PRJNA1104541/SRR28808391/SRR28808391_annotated.vcf\n",
      "/home/louie/Projects/Personal/GraphDB/Data/GWAS/PRJNA1104541/SRR28808372/SRR28808372_annotated.vcf\n",
      "DF generation complete | 0.677 minutes\n"
     ]
    }
   ],
   "source": [
    "maindir   = datadir+\"/GWAS/PRJNA1104541/\"\n",
    "df        = pd.DataFrame(columns = [\"Chr\", \"Pos\",\"Ref\", \"Alt\",\"Gene\",\"Effect\",\"Impact\",\"Transcript\",\"ProteinChange\"])\n",
    "\n",
    "start = time.time()\n",
    "for path, subdirs, files in os.walk(maindir):\n",
    "    for name in files:\n",
    "        if \"annotated.vcf\" in name:\n",
    "            vcf_file = os.path.join(path, name)\n",
    "            print(vcf_file)\n",
    "            records = []\n",
    "            for variant in VCF(vcf_file):\n",
    "                ann_field = variant.INFO.get(\"ANN\")\n",
    "                for ann in ann_field.split(\",\"):\n",
    "                    fields = ann.split(\"|\")\n",
    "                    records.append({\n",
    "                        \"Chr\": variant.CHROM,\n",
    "                        \"Pos\": variant.POS,\n",
    "                        \"Ref\": variant.REF,\n",
    "                        \"Alt\": \",\".join(variant.ALT),\n",
    "                        \"Gene\": fields[3],\n",
    "                        \"Effect\": fields[1],\n",
    "                        \"Impact\": fields[2],\n",
    "                        \"Transcript\": fields[6],\n",
    "                        \"ProteinChange\": fields[10]\n",
    "                    })\n",
    "            tmpdf = pd.DataFrame(records)\n",
    "            df = pd.concat([df, tmpdf.loc[tmpdf[\"Impact\"] == \"HIGH\"]])\n",
    "print(f'VCF parsing complete | {round((time.time()-start)/60, 3)} minutes')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "17bc2bd9-d266-4bfa-a857-f3f4ea79b4c2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['SDF4', 'SPEN', 'RPL11', 'TRNAU1AP', 'EPHA10', 'MACF1', 'NRDC',\n",
       "       'PLPP3', 'SRSF11', 'RPL5', 'DDX20', 'S100A9', 'S100A8', 'KRTCAP2',\n",
       "       'ASH1L', 'IFI16', 'FMO2', 'CEP350', 'LAMC2', 'LBR', 'GUK1',\n",
       "       'SDCCAG8', 'AKT3', 'OPTN', 'MLLT10', 'HNRNPH3', 'CCAR1', 'ADIRF',\n",
       "       'BORCS7-ASMT', 'AS3MT', 'GSTO1', 'LOC124902554', 'SEC23IP',\n",
       "       'ACADSB', 'IFITM3', 'EPS8L2', 'RPL27A', 'RPS13', 'PEX16', 'PSMC3',\n",
       "       'NDUFS3', 'SLC15A3', 'WDR74', 'FAU', 'YIF1A', 'RNF169', 'FAT3',\n",
       "       'CWF19L2', 'NNMT', 'ATP5MG', 'KMT2A', 'KCNJ1', 'WNK1', 'NDUFA9',\n",
       "       'PTMS', 'A2M', 'PLBD1', 'MYG1', 'BAZ2A', 'CNOT2', 'ATP2B1',\n",
       "       'UHRF1BP1L', 'HSP90B1', 'SART3', 'ALKBH2', 'CLIP1', 'ZNF891',\n",
       "       'POMP', 'MRPL52', 'PSME1', 'BAZ1A', 'NEMF', 'KTN1', 'DAAM1',\n",
       "       'ZBTB25', 'RBM25', 'AREL1', 'SLIRP', 'CALM1', 'IGHGP', 'INO80',\n",
       "       'DUT', 'ZNF609', 'RPLP1', 'MYO9A', 'UBE2Q2', 'SCAPER', 'PSMA4',\n",
       "       'AKAP13', 'MCTP2', 'HBA1', 'METTL26', 'TSC2', 'SRRM2', 'ELOB',\n",
       "       'SMIM22', 'UBN1', 'MYH11', 'IGSF6', 'TNRC6A', 'SULT1A2', 'MVP',\n",
       "       'KIFC3', 'LRRC36', 'ZFHX3', 'CPNE7', 'TMEM102', 'TMEM220', 'RPL19',\n",
       "       'MED1', 'KRT19', 'AARSD1', 'PTGES3L-AARSD1', 'CCDC43', 'NMT1',\n",
       "       'LUC7L3', 'DHX40', 'PSMC5', 'USP36', 'ASPSCR1', 'PIEZO2', 'ATP8B1',\n",
       "       'CDC34', 'POLRMT', 'ZNF561-AS1', 'SMARCA4', 'BRD4', 'MAP1S',\n",
       "       'PGPEP1', 'UBA52', 'TMEM147', 'ETV2', 'HCST', 'COX7A1', 'FBL',\n",
       "       'ZNF546', 'PRX', 'SULT2B1', 'RPL18', 'NAPSA', 'CHMP2A', 'IAH1',\n",
       "       'PRR30', 'IFT172', 'SPDYA', 'CEP68', 'WDR54', 'CHMP3',\n",
       "       'RNF103-CHMP3', 'MRPS5', 'ZC3H6', 'DYNC1I2', 'TTN', 'ZC3H15',\n",
       "       'TFPI', 'WDR75', 'RPL37A', 'GLB1L', 'TUBA4A', 'CPNE1', 'ROMO1',\n",
       "       'CTNNBL1', 'STK4', 'PSMA7', 'RPS21', 'CFAP298-TCP10L', 'CFAP298',\n",
       "       'SON', 'ATF4', 'NCAPH2', 'SCO2', 'ODF3B', 'BHLHE40', 'IL17RC',\n",
       "       'EMC3', 'TAMM41', 'EXOG', 'DALRD3', 'CCDC80', 'CFAP44', 'CSTA',\n",
       "       'PARP9', 'MBD4', 'ASTE1', 'CEP63', 'BOD1L1', 'DCAF16', 'FAM114A1',\n",
       "       'SGCB', 'SRP72', 'AFF1', 'TTC23L', 'NIPBL', 'FYB1', 'DAB2', 'CAST',\n",
       "       'HSPA4', 'TTC1', 'MRNIP', 'RACK1', 'MYLIP', 'BTN3A3', 'LST1',\n",
       "       'PSMB9', 'CUTA', 'HSP90AB1', 'PHF3', 'TENT5A', 'ZNF292', 'RARS2',\n",
       "       'UBE2J1', 'PNISR', 'RNF146', 'ADGRG6', 'EZR', 'MAP3K4', 'RNASET2',\n",
       "       'SNX8', 'TAX1BP1', 'ERV3-1', 'PTPN12', 'KMT2E', 'CALD1', 'KMT2C',\n",
       "       'TACC1', 'ADAM9', 'PCMTD1', 'COPS5', 'GRHL2', 'RIC1', 'KIAA2026',\n",
       "       'PLIN2', 'UNC13B', 'CCDC107', 'ALDH1A1', 'PHF2', 'CLIC3', 'TRNM',\n",
       "       'TRNW', 'COX1', 'COX2', 'ATP8', 'COX3', 'ND4L', 'ND4', 'ND5',\n",
       "       'TRNP', 'HCCS', 'PRDX4', 'MED14', 'CFP', 'HSD17B10', 'NONO',\n",
       "       'PIN4', 'ATRX', 'DIAPH2', 'BEX1'], dtype=object)"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[\"Gene\"].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "aec5669e-7d53-4995-88b3-304cda623e4e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "frameshift_variant\n",
      "stop_lost\n",
      "frameshift_variant&stop_lost\n",
      "frameshift_variant&splice_region_variant\n",
      "stop_gained\n",
      "frameshift_variant&stop_gained\n",
      "start_lost\n",
      "stop_gained&splice_region_variant\n",
      "stop_lost&conservative_inframe_deletion\n",
      "frameshift_variant&splice_acceptor_variant&splice_region_variant&intron_variant\n",
      "start_lost&disruptive_inframe_deletion\n",
      "splice_acceptor_variant&conservative_inframe_deletion&splice_region_variant&intron_variant\n",
      "frameshift_variant&splice_donor_variant&splice_region_variant&intron_variant\n",
      "frameshift_variant&start_lost\n",
      "frameshift_variant&stop_gained&splice_region_variant\n",
      "frameshift_variant&stop_lost&splice_region_variant\n",
      "start_lost&splice_region_variant\n",
      "stop_lost&disruptive_inframe_deletion\n",
      "splice_donor_variant&conservative_inframe_deletion&splice_region_variant&intron_variant\n"
     ]
    }
   ],
   "source": [
    "for i in df.loc[df[\"ProteinChange\"] != \"\"][\"Effect\"].unique():\n",
    "    print(i)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "71e77b21-95fd-43be-94f5-538d8cd190ab",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Pos</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Effect</th>\n",
       "      <th>Impact</th>\n",
       "      <th>Transcript</th>\n",
       "      <th>ProteinChange</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>626</th>\n",
       "      <td>1</td>\n",
       "      <td>1652532</td>\n",
       "      <td>GT</td>\n",
       "      <td>G</td>\n",
       "      <td>CDK11B</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>XM_011542490.4</td>\n",
       "      <td>p.Gln89fs</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>627</th>\n",
       "      <td>1</td>\n",
       "      <td>1652532</td>\n",
       "      <td>GT</td>\n",
       "      <td>G</td>\n",
       "      <td>CDK11B</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>XM_011542494.4</td>\n",
       "      <td>p.Gln55fs</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>628</th>\n",
       "      <td>1</td>\n",
       "      <td>1652532</td>\n",
       "      <td>GT</td>\n",
       "      <td>G</td>\n",
       "      <td>CDK11B</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>XM_017002928.3</td>\n",
       "      <td>p.Gln55fs</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>629</th>\n",
       "      <td>1</td>\n",
       "      <td>1652532</td>\n",
       "      <td>GT</td>\n",
       "      <td>G</td>\n",
       "      <td>CDK11B</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>XM_047435355.1</td>\n",
       "      <td>p.Gln55fs</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>630</th>\n",
       "      <td>1</td>\n",
       "      <td>1652532</td>\n",
       "      <td>GT</td>\n",
       "      <td>G</td>\n",
       "      <td>CDK11B</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>XM_047435360.1</td>\n",
       "      <td>p.Gln55fs</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2727828</th>\n",
       "      <td>X</td>\n",
       "      <td>149490351</td>\n",
       "      <td>CA</td>\n",
       "      <td>CAA</td>\n",
       "      <td>IDS</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>NM_000202.8</td>\n",
       "      <td>p.Ala324fs</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2727829</th>\n",
       "      <td>X</td>\n",
       "      <td>149490351</td>\n",
       "      <td>CA</td>\n",
       "      <td>CAA</td>\n",
       "      <td>IDS</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>NM_001166550.4</td>\n",
       "      <td>p.Ala234fs</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2727830</th>\n",
       "      <td>X</td>\n",
       "      <td>149490351</td>\n",
       "      <td>CA</td>\n",
       "      <td>CAA</td>\n",
       "      <td>IDS</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>NM_006123.5</td>\n",
       "      <td>p.Ala324fs</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2729534</th>\n",
       "      <td>X</td>\n",
       "      <td>153935301</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>RENBP</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>NM_002910.6</td>\n",
       "      <td>p.Cys423*</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2729535</th>\n",
       "      <td>X</td>\n",
       "      <td>153935301</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>RENBP</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>XM_017029698.2</td>\n",
       "      <td>p.Cys413*</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>65414 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Chr        Pos Ref  Alt    Gene              Effect Impact  \\\n",
       "626       1    1652532  GT    G  CDK11B  frameshift_variant   HIGH   \n",
       "627       1    1652532  GT    G  CDK11B  frameshift_variant   HIGH   \n",
       "628       1    1652532  GT    G  CDK11B  frameshift_variant   HIGH   \n",
       "629       1    1652532  GT    G  CDK11B  frameshift_variant   HIGH   \n",
       "630       1    1652532  GT    G  CDK11B  frameshift_variant   HIGH   \n",
       "...      ..        ...  ..  ...     ...                 ...    ...   \n",
       "2727828   X  149490351  CA  CAA     IDS  frameshift_variant   HIGH   \n",
       "2727829   X  149490351  CA  CAA     IDS  frameshift_variant   HIGH   \n",
       "2727830   X  149490351  CA  CAA     IDS  frameshift_variant   HIGH   \n",
       "2729534   X  153935301   G    T   RENBP         stop_gained   HIGH   \n",
       "2729535   X  153935301   G    T   RENBP         stop_gained   HIGH   \n",
       "\n",
       "             Transcript ProteinChange  \n",
       "626      XM_011542490.4     p.Gln89fs  \n",
       "627      XM_011542494.4     p.Gln55fs  \n",
       "628      XM_017002928.3     p.Gln55fs  \n",
       "629      XM_047435355.1     p.Gln55fs  \n",
       "630      XM_047435360.1     p.Gln55fs  \n",
       "...                 ...           ...  \n",
       "2727828     NM_000202.8    p.Ala324fs  \n",
       "2727829  NM_001166550.4    p.Ala234fs  \n",
       "2727830     NM_006123.5    p.Ala324fs  \n",
       "2729534     NM_002910.6     p.Cys423*  \n",
       "2729535  XM_017029698.2     p.Cys413*  \n",
       "\n",
       "[65414 rows x 9 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "display(df.loc[df[\"ProteinChange\"] != \"\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "8004def9-69df-4667-9190-a57eb3b62c2d",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "18660"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(df.loc[df[\"ProteinChange\"] != \"\"][\"ProteinChange\"].unique())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "d81dbd97-cdc6-4bc0-b9db-fa8c4ef7584f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Effect</th>\n",
       "      <th>Impact</th>\n",
       "      <th>Transcript</th>\n",
       "      <th>ProteinChange</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>630665</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>LOC101928626</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>NR_125957.1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>630665</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>MIR12136</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>NR_162149.1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>630665</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>MIR12136</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>unassigned_transcript_4</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>630665</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>MTND2P28</td>\n",
       "      <td>intragenic_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>MTND2P28</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>630668</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>LOC101928626</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>NR_125957.1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>95</th>\n",
       "      <td>1</td>\n",
       "      <td>1218635</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>TNFRSF4</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>XM_011542076.3</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96</th>\n",
       "      <td>1</td>\n",
       "      <td>1218635</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>TNFRSF4</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>NM_003327.4</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>1</td>\n",
       "      <td>1218635</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>TNFRSF4</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>XM_011542075.3</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>98</th>\n",
       "      <td>1</td>\n",
       "      <td>1218635</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>TNFRSF4</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>XR_007063145.1</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>99</th>\n",
       "      <td>1</td>\n",
       "      <td>1218635</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>SDF4</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>XM_011541556.2</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>100 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   CHROM      POS REF ALT          Gene                   Effect    Impact  \\\n",
       "0      1   630665   T   C  LOC101928626    upstream_gene_variant  MODIFIER   \n",
       "1      1   630665   T   C      MIR12136  downstream_gene_variant  MODIFIER   \n",
       "2      1   630665   T   C      MIR12136  downstream_gene_variant  MODIFIER   \n",
       "3      1   630665   T   C      MTND2P28       intragenic_variant  MODIFIER   \n",
       "4      1   630668   G   A  LOC101928626    upstream_gene_variant  MODIFIER   \n",
       "..   ...      ...  ..  ..           ...                      ...       ...   \n",
       "95     1  1218635   T   C       TNFRSF4    upstream_gene_variant  MODIFIER   \n",
       "96     1  1218635   T   C       TNFRSF4    upstream_gene_variant  MODIFIER   \n",
       "97     1  1218635   T   C       TNFRSF4    upstream_gene_variant  MODIFIER   \n",
       "98     1  1218635   T   C       TNFRSF4    upstream_gene_variant  MODIFIER   \n",
       "99     1  1218635   T   C          SDF4  downstream_gene_variant  MODIFIER   \n",
       "\n",
       "                 Transcript ProteinChange  \n",
       "0               NR_125957.1                \n",
       "1               NR_162149.1                \n",
       "2   unassigned_transcript_4                \n",
       "3                  MTND2P28                \n",
       "4               NR_125957.1                \n",
       "..                      ...           ...  \n",
       "95           XM_011542076.3                \n",
       "96              NM_003327.4                \n",
       "97           XM_011542075.3                \n",
       "98           XR_007063145.1                \n",
       "99           XM_011541556.2                \n",
       "\n",
       "[100 rows x 9 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "from cyvcf2 import VCF\n",
    "\n",
    "testvcf = \"/home/louie/Projects/Personal/GraphDB/Data/GWAS/PRJNA1104541/SRR28808386/SRR28808386_annotated.vcf\"\n",
    "\n",
    "records = []\n",
    "for variant in VCF(testvcf):\n",
    "    ann_field = variant.INFO.get(\"ANN\")\n",
    "    #if not ann_field:\n",
    "    #    continue\n",
    "    for ann in ann_field.split(\",\"):\n",
    "        fields = ann.split(\"|\")\n",
    "        records.append({\n",
    "            \"CHROM\": variant.CHROM,\n",
    "            \"POS\": variant.POS,\n",
    "            \"REF\": variant.REF,\n",
    "            \"ALT\": \",\".join(variant.ALT),\n",
    "            \"Gene\": fields[3],\n",
    "            \"Effect\": fields[1],\n",
    "            \"Impact\": fields[2],\n",
    "            \"Transcript\": fields[6],\n",
    "            \"ProteinChange\": fields[10]\n",
    "        })\n",
    "\n",
    "df = pd.DataFrame(records)\n",
    "display(df.head(100))\n",
    "\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
